pkg/registry/core/pod/storage/eviction.go

```diff
@@ -324,13 +343,25 @@ func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string,
 			return podObj, nil
 		}
 
-		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, conditionAppender)
+		// pass the potentially stale pod here only to trigger the default UID precondition
+		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, setPreconditions, conditionAppender) // order important
```

Comment:
pass in `nil` for the new pod, `setPreconditions` is throwing away the newObj so it's not used at all

Reply:
I was mostly concerned about the case if pod and its uid changes midflight since we last obtained it from get - should we fail on that? But can change this to nil, if this isn't a real concern.

Reply:
> I was mostly concerned about the case if pod and its uid changes midflight since we last obtained it from get

Yeah, go ahead and change this to nil. if the uid/resourceVersion matching is important, the passed-in DeleteOptions will tell us that, and we can honor those using our `podWithPreconditions` transformer

Reply:
ok, changed this to nil. And let's honor only the delete options